cou/utils/juju_utils.py

```diff
@@ -253,17 +254,18 @@ async def async_get_unit_from_name(
     try:
         if model is None:
             model = await _async_get_model(model_name)
+
         units = model.applications[app].units
     except KeyError as exc:
-        msg = f"Application: {app} does not exist in current model"
-        logger.error(msg)
-        raise UnitNotFound(unit_name) from exc
+        raise UnitNotFound(f"Application {app} not found in model.") from exc
```

Comment:
We know the model name so why not:
```suggestion
        raise UnitNotFound(f"Application {app} not found in model {model_name}.") from exc
```

Reply:
I didn't use `model_name` here because `model_name` is an optional input. I have re-arranged the function a bit to move model object assignment outside the try-except block (we were not trying to catch any exception for this part anyway). This way we can simply call `model.name` to get the actual name.